### PR TITLE
Holotool refactor

### LINKED ISSUE #8960: Cant put holotool on belt or backpack after switching to tool
[Round ID]: 31182
[Reproduction]: use the holotool as a tool. try and put it in belt or back pack. will say that its stuck to your hand even tho you can drop it and such

### PR DESCRIPTION
**Pull request by ghost**
:cl: steamp0rt
refactor: Refactored holotool
fix: Holotool should no longer get stuck in your hand
tweak: Holo-knife now does slightly more damage (17, up from 5)
tweak: Holo-knife now can penetrate armor well
tweak: Holo-knife has a chance to embed when thrown.
add: Holotools now have radial menus, Ctrl+Click on them to open it!
/:cl:

capture

WARNING: WILL NOT WORK ON ITEMS THAT DO NOT USE TOOL BEHAVIOR TO DETECT TOOLS

Fixes #8960

### CODE REVIEW

**Comment by Carbonhell** [comment]
The tg changes should be PRed to tg. Since you are gitbanned i could do it.

**Comment by GuyonBroadway** [comment]
Make sure to set the toolspeed to 0.2ish cus all tools were made much faster in a recent rebalance and the holotool was overlooked making it objectivly worthless.

**Comment by ghost** [comment]
@Carbonhell Sure. Yes, I did test them.

**Comment by ghost** [comment]
stupid travis stalling

**Comment by ghost** [comment]
capture
Added radial menu

File: hippiestation/code/game/objects/items/holotool/modes.dm [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+	name = "holo-knife"
+	sound = 'sound/weapons/blade1.ogg'
+
+/datum/holotool_mode/knife/can_be_used(var/obj/item/holotool/H)
+	return (H.obj_flags & EMAGGED)
+
+/datum/holotool_mode/knife/on_set(var/obj/item/holotool/H)
+	..()
+	H.sharpness = IS_SHARP
+	H.force = 9
```
Comment (GuyonBroadway) [inline hippiestation/code/game/objects/items/holotool/modes.dm]: Just saying, 9 is a bit shit when the welder itself does 15 burn damage a pop when lit.

**Comment by ghost** [comment]
capture

That looks better.

File: hippiestation/code/game/objects/items/holotool/modes.dm (hunk 1 above)
Comment (ghost) [reply hippiestation/code/game/objects/items/holotool/modes.dm]: What should it be, then?

Comment (GuyonBroadway) [reply hippiestation/code/game/objects/items/holotool/modes.dm]: Considering it needs an emag and for 2tc you can get an edagger that does 18 with similar armour pen, 12-15 is not unreasonable.

Consider toolboxes and fire extinguishers do 12 and circ saws which can be printed from the autolathe for cheap do 15.

**Comment by GuyonBroadway** [comment]
17 is respectable. It can be nerfed if need be

File: hippiestation/code/game/objects/items/holotool/modes.dm [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	name = "holo-knife"
+	sound = 'sound/weapons/blade1.ogg'
+
+/datum/holotool_mode/knife/can_be_used(var/obj/item/holotool/H)
+	return (H.obj_flags & EMAGGED)
+
+/datum/holotool_mode/knife/on_set(var/obj/item/holotool/H)
+	..()
+	H.sharpness = IS_SHARP
+	H.force = 17
```
Comment (yoyobatty) [inline hippiestation/code/game/objects/items/holotool/modes.dm:64]: A round start 17 damage knife... I'm sorry, what?

File: hippiestation/code/game/objects/items/holotool/modes.dm [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	H.attack_verb = list("sliced", "torn", "cut")
+	H.armour_penetration = 45
```
Comment (yoyobatty) [inline hippiestation/code/game/objects/items/holotool/modes.dm:66]: This is not going into the game.

**Review by yoyobatty — DISMISSED** [review]
That knife is too strong. All of you powergamers can fuck off.

File: hippiestation/code/game/objects/items/holotool/modes.dm (hunk 2 above)
Comment (ghost) [reply hippiestation/code/game/objects/items/holotool/modes.dm:64]: Not roundstart. Only when emagged.

File: hippiestation/code/game/objects/items/holotool/modes.dm (hunk 3 above)
Comment (ghost) [reply hippiestation/code/game/objects/items/holotool/modes.dm:66]: Not roundstart. Only when emagged.

**Comment by ghost** [comment]
Dis is merge-ready. Not sure if this is big enough to need the +3 or not.

**Comment by Eagle-Eyess** [comment]
+1. Also 17 dmg isn’t that much compared to others like the energy sword.

**Comment by ghost** [comment]
tweak: Holo-knife now does slightly more damage (17, up from 5)
yeah nice, slightly. it's good though

**Comment by GuyonBroadway** [comment]
The holo knife requires you to emag the holotool which requires you to spend 6tc when for 3tc I can get a 20 damage combat knife or for 2tc an 18 damage e-dagger

File: hippiestation/code/game/objects/items/holotool/holotool.dm [hunk 4]
```diff
@@ -0,0 +1,118 @@
+
+
```
Comment (praisenarsie) [inline hippiestation/code/game/objects/items/holotool/holotool.dm]: whitespace

**Review by praisenarsie — DISMISSED** [review]
seems fine

**Comment by ghost** [comment]
there's your damn whitespace fixed

**Review by yoyobatty — APPROVED** [review]
Needs 2 +1's from headmin @Carbonhell @t-f-w @flux

**Comment by Carbonhell** [comment]
Don't bundle refactors with features. It's extremely frowned upon.
I do not like the knife. This is supposed to be an useful tool, giving the RD just a slightly better edagger seems pretty shit. I'd rather prefer some sort of tool given to him. -1
If possible, strip the knife altogether so the PR can be merged as a fix.

**Comment by GuyonBroadway** [comment]
Three points carbon

1: the knife was always part of the holotool
2: you need an emag to unlock it
3: its weaker than an edagger doing 17 damage as opposed to 18 with similar armour pen.

**Comment by Carbonhell** [comment]
It was already in?the fuck...huh,alright then. regardless,meh.

**Comment by Carbonhell** [comment]
Alright, +1. @fluxcapacitor1337

**Comment by ghost** [comment]
+1

**Comment by earthmoonjune** [comment]
+1

**Comment by GuyonBroadway** [comment]
Well that's at least three head +1's, I'll assume they are for the holo-knife changes and woe betide me otherwise.

**Pull request merged**